lib/net/http_xhr_plugin.js

```diff
@@ -39,7 +39,11 @@ shaka.net.HttpXHRPlugin = class {
 
     const promise = new Promise(((resolve, reject) => {
       xhr.open(request.method, uri, true);
-      xhr.responseType = 'arraybuffer';
+
+      if (requestType != shaka.net.NetworkingEngine.RequestType.LICENSE) {
```

Comment:
⚠️ TO BE CHANGED ⚠️ 

❓ 
When requesting the license server with a XHR responseType `arrayBuffer` (with fetch or XHR), I have a `Type error` when updating the EME session. When removing the responseType, I get a base64 string from my license server which I am able then to transform into an ArrayBuffer.

The values are completely different when receiving a ArrayBuffer from `xhr.response` or when creating the ArrayBuffer from `xhr.responseText`, if you guys have any idea on this, it would be appreciated (maybe a wrong server implementation, but I doubt that ?) !

So far, to make it work for me, I had to write that code sample ...

Reply:
Resolved: I add to add a special data treatment in shaka's responseFilter to have a correctly formated data:

```javascript
if (keySystem === 'com.apple.fps') {
  const dataView = new DataView(response.data);
  const decoder = new TextDecoder('utf-8');

  response.data = shaka.util.Uint8ArrayUtils.fromBase64(decoder.decode(dataView));
}
```